component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -336,6 +327,135 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         }
     }
 
+    /**
+     * Process valid user token.
+     *
+     * @param context           AuthenticationContext.
+     * @param authenticatedUser AuthenticatedUser.
+     */
+    private void processValidUserToken(AuthenticationContext context, AuthenticatedUser authenticatedUser) {
+
+        context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, EmailOTPAuthenticatorConstants.EMPTY_STRUNG);
+        context.setProperty(EmailOTPAuthenticatorConstants.EMAILOTP_ACCESS_TOKEN,
+                EmailOTPAuthenticatorConstants.EMPTY_STRUNG);
+        context.setSubject(authenticatedUser);
+        String emailFromProfile = context.getProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL).toString();
```

Comment:
shall we add a null check on context.getProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL)

Reply:
Added